.github/actions/next-stats-action/src/index.js

```diff
@@ -105,9 +105,12 @@ if (!allowedActions.has(actionInfo.actionName) && !actionInfo.isRelease) {
     for (const dir of repoDirs) {
       logger(`Running initial build for ${dir}`)
       if (!actionInfo.skipClone) {
+        const usePnpm = await fs.pathExists(path.join(dir, 'pnpm-lock.yaml'))
         let buildCommand = `cd ${dir}${
           !statsConfig.skipInitialInstall
-            ? ' && yarn install --network-timeout 1000000'
+            ? usePnpm
+              ? ' && pnpm install && pnpm run build'
+              : ' && yarn install --network-timeout 1000000'
```

Comment:
Is it expected that pnpm also runs build but yarn doesn't? Is this related to the way prepublish step is executed?

Reply:
Yeah `yarn` automatically calls `prepublish` after install, we can also remove this check for `yarn` once PRs get updated to use `pnpm` instead.

Reply:
We should probably change `prepublish` to `prepublishOnly` so it works the same for all package managers

Reply:
Sounds good, updated in https://github.com/vercel/next.js/pull/37259/commits/41c8f8f8e8dd13acac9c5de393f0829b7e475bb7